src/containers/Task/Task.jsx

```diff
@@ -2,139 +2,102 @@ import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { useHistory } from 'react-router-dom';
 import { gql } from 'apollo-boost';
-import client from '../../graphql';
+import { useLazyQuery } from '@apollo/react-hooks';
 import { enhanceUrlWithParameters, getCampaignDigitalDocument } from '../../utils';
 import ActiveTask from '../../components/ActiveTask';
 import { NotificationContext } from '../NotificationsProvider/NotificationsProvider';
 import { GET_CAMPAIGN } from '../../screens/ActiveCampaign/';
 
 export default function Task({ campaignIdentifier, taskIdentifier }) {
-  const history                               = useHistory();
-  const [loading, setLoading]                 = useState(false);
-  const [campaign, setCampaign]               = useState(null);
-  const [task, setTask]                       = useState(null);
-  const [taskIdentifiers, setTaskIdentifiers] = useState([]);
-  const [tasksReceived, setTasksReceived]     = useState([]);
-  const [taskTotal, setTaskTotal]             = useState(10);
-  const taskDrawAmount                        = 10;
-  const nickname                              = localStorage.getItem('nickname');
-
-  const onGoBackClick = () => {
-    history.goBack();
-  };
-
-  const handleNextTaskButtonClick = () => {
-    const remainingTaskIdentifiers = taskIdentifiers.filter(identifier => identifier !== taskIdentifier);
-    const hasReceivedTask          = tasksReceived.includes(taskIdentifier);
-
-    if (!hasReceivedTask) {
-      setTasksReceived(tasksReceived.concat(taskIdentifier));
-    }
-
-    setTaskIdentifiers(remainingTaskIdentifiers);
-    if (remainingTaskIdentifiers.length > 0) {
-      history.replace(`/campaign/${campaignIdentifier}/task/${remainingTaskIdentifiers[0]}`);
+  const history                             = useHistory();
+  const [campaign, setCampaign]             = useState(null);
+  const [task, setTask]                     = useState(null);
+  const [tasksTodo, setTasksTodo]           = useState([]);
+  const [tasksSubmitted, setTasksSubmitted] = useState([]);
+  const taskDrawAmount                      = 10;
+  const refetchTimeout                      = 3000;
```

Comment:
Is this not a bit much? do we expect new tasks to be added every 3 seconds?

Reply:
That was for testing purposes, but thanks for reminding me. 30000 it is!